databuilder/databuilder/extractor/redshift_table_usage_extractor.py

```diff
@@ -0,0 +1,44 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+
+from databuilder.extractor.base_postgres_column_usage_extractor import BasePostgresColUsageExtractor
+
+
+class RedshiftTableUsagextractor(BasePostgresColUsageExtractor):
+    """
+    Extracts Redshift table usage stats from underlying meta store database using SQLAlchemyExtractor
+    This differs from the PostgresMetadataExtractor because in order to support Redshift's late binding views,
+    we need to join the INFORMATION_SCHEMA data against the function PG_GET_LATE_BINDING_VIEW_COLS().
+    """
+
+    def get_sql_statement(self, use_catalog_as_cluster_name: bool, where_clause_suffix: str) -> str:
+        if use_catalog_as_cluster_name:
+            cluster_source = "CURRENT_DATABASE()"
+        else:
+            cluster_source = f"'{self._cluster}'"
+
+        return """
+        select
+            'redshift' as database,
+            {cluster_source} as cluster,
+            trim(sti.schema) as schema,
+            trim(sti.table) as table_name,
+            trim(pu.usename) as user_email,
+             count(sq.query) as read_count
+            from
+            (select
+                distinct userid, tbl , query
+            from stl_scan) ss
+        left join SVV_TABLE_INFO sti on sti.table_id = ss.tbl
+        left join pg_user pu on pu.usesysid = ss.userid
+        left join stl_query sq on ss.query = sq.query
```

Comment:
the `stl_query` table has a max of ~4k characters per record and queries longer than 4k chars need to concat all of the results together to get the full, original query. For example, if a query is 10k characters long it will return 3 records. This will impact the `read count`. You may want to build some form of query uniqueness as a subquery before counting